frontend/packages/console-dynamic-plugin-sdk/src/extensions/navigation.ts

```diff
@@ -0,0 +1,69 @@
+import { Extension } from '@console/plugin-sdk/src/typings/base';
+
+export namespace ExtensionProperties {
+  export type NavItem = {
+    /** A unique identifier for this item. */
+    id: string;
+    /** The perspective ID to which this item belongs to. If not specified, contributes to the default perspective. */
+    perspective?: string;
+    /** Navigation section to which this item belongs to. If not specified, render this item as a top level link. */
+    section?: string;
+    /** Mark this item as active when the URL starts with one of these paths. */
+    startsWith?: string[];
+    /** Insert this item before the item referenced here. */
+    insertBefore?: string;
```

Comment:
We should align with the new capabilities for `insertBefore` and `insertAfter`
```suggestion
    /** Insert this item before the item referenced here. For arrays, the first one found in order is used. */
    insertBefore?: string | string[];
    /** Insert this item after the item referenced here. For arrays, the first one found in order is used. `insertBefore` takes precedence. */
    insertAfter?: string | string[];
```